framework/web/Request.php

```diff
@@ -168,45 +169,46 @@ class Request extends \yii\base\Request
     /**
      * @var array the configuration for trusted security related headers.
      *
-     * An array key is a regular expression for matching a client (ususally a proxy) by
-     * host name or ip address.
+     * An array key is an IPv4 or IPv6 IP address in CIDR notation for matching a client.
      *
      * An array value is a list of headers to trust. These will be matched against
      * [[secureHeaders]] to determine which headers are allowed to be sent by a specified host.
      * The case of the header names must be the same as specified in [[secureHeaders]].
      *
-     * To specify a host for which you trust all headers listed in [[secureHeaders]] you can just specify
-     * the regular expression as the array value.
-     *
-     * For example, to trust all headers listed in [[secureHeaders]]
-     * from domains ending in '.trusted.com' use the following:
+     * For example, to trust all headers listed in [[secureHeaders]] for IP addresses
+     * in range `192.168.0.0-192.168.0.254` write the following:
      *
      * ```php
      * [
-     *     '/\.trusted\.com$/',
+     *     '192.168.0.0/24',
      * ]
      * ```
      *
-     * To trust just the `x-forwarded-for` header from domains ending in `.partial.com` use:
+     * To trust just the `X-Forwarded-For` header from `10.0.0.1`, use:
      *
      * ```
      * [
-     *     '/\.partial\.com$/' => ['X-Forwarded-For']
+     *     '10.0.0.1' => ['X-Forwarded-For']
      * ]
      * ```
      *
      * Default is to trust all headers except those listed in [[secureHeaders]] from all hosts.
-     * Matches are tried in order and searching is stopped when a host or IP matches.
+     * Matches are tried in order and searching is stopped when IP matches.
+     * 
+     * > Info: Matching is performed using [[yii\validators\IpValidator|IpValidator]]. 
+     *   See [[yii\validators\IpValidator::networks|IpValidator::setRanges()]] 
+     *   and [[yii\validators\IpValidator::networks|IpValidator::networks]] for advanced matching.
+     * 
      * @see $secureHeaders
      * @since 2.0.13
      */
-    public $trustedHosts = [];
+    public $trustHeadersFrom = [];
```

Comment:
👎 for name change - previous was IMO much better.

Reply:
Yeah, this one is disputable. I have decided to change the name because it's not about hostname anymore - rather IP address or subnets. However, `trustHeadersFrom` is not very good because it does not says FROM WHAT.

:+1: for `trustedHosts`
:laughing: for `trustHeadersFrom`

p.s.: According to wiki definition of word [host](https://en.wikipedia.org/wiki/Host_(network)), it can be used.

Reply:
> I have decided to change the name because it's not about hostname anymore

Old name newer mentions "hostname". Host can be identified by IP.

Also I prefer using nouns for properties.

Reply:
> because it's not about hostname anymore

its does not say trusted Host**names** but just trusted hosts ;)

Reply:
Thank you for discussion, friends